app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt

```diff
@@ -854,16 +855,27 @@ abstract class BaseBrowserFragment :
         }
     }
 
+    private val startForResult = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) {
+            result: ActivityResult ->
+        if (result.resultCode == Activity.RESULT_OK) {
```

Comment:
Seeing that this is declared in multiple files I'm thinking that it can be extracted to an extension function that can then be used in all places? Something like:
```kotlin
fun Fragment.startForResult(intent: Intent, onSuccess: () -> Unit) {
    registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
        if (result.resultCode == Activity.RESULT_OK) {
            onSuccess()
        }
    }.launch(intent)
}
```